eng/testing/xunit/xunit.targets

```diff
@@ -6,6 +6,11 @@
                       Condition="'$(TargetFrameworkIdentifier)' == '.NETCoreApp'" />
   </ItemGroup>
 
+  <PropertyGroup Condition="'$(TargetFrameworkIdentifier)' == '.NETFramework'">
+    <AutoGenerateBindingRedirects Condition="'$(AutoGenerateBindingRedirects)' == ''">true</AutoGenerateBindingRedirects>
+    <GenerateBindingRedirectsOutputType Condition="'$(GenerateBindingRedirectsOutputType)' == ''">true</GenerateBindingRedirectsOutputType>
+  </PropertyGroup>
+
```

Comment:
In the main PR I found no description for this change except "Fixing binding redirects for remote excutor". For my own education, would you mind explaining this?

Reply:
We weren't generating any bindingRedirects for tests, but they need them.

When running from XUnit we were working because XUnit has an AssemblyResolve hook that does a similar thing (though it can't do the exact same thing).

When I added out-of-proc services with RemoteExecutor -- really any use of remote executor -- loads failed since they didn't have this resolve handler.  Ensuring the app.config is generated makes sure that remote executor gets it.  It will also avoid us needing to rely on the AssemblyResolve hook in Xunit (and it's fragility).

Here's the discussion: https://github.com/dotnet/runtime/pull/83892#issuecomment-1492189875